### PR TITLE
feat(braze): allowing braze content proxy to access lists

### PR DESCRIPTION
**Pull request by bassrock**
## Goal

Enable future work for digest emails

# Accomplishes
* Moves JWT generation to it's own package
* Adds a digest endpoint that allows Braze to get a digest of data for a user for email or push. Right now its just the most recent 3 saves

https://mozilla-hub.atlassian.net/browse/POCKET-10480

### CODE REVIEW

File: servers/braze-content-proxy/src/routes/digest.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+): BrazeSavedItem[] {
+  const edges = response.data.user?.savedItems?.edges;
+  if (!edges) {
+    return [];
+  }
+  const savedItems: BrazeSavedItem[] = edges
+    .filter((edge) => edge.node.item.__typename == 'Item')
+    .map((edge) => {
+      const item = edge.node.item;
+      if (item.__typename == 'Item') {
```
Comment (kschelonka) [inline servers/braze-content-proxy/src/routes/digest.ts:46]: is the typescript compiler not inferring this?

File: servers/braze-content-proxy/src/routes/scheduledItems.integration.ts [hunk 2]
```diff
@@ -65,15 +65,14 @@ describe('/scheduled-items/:scheduledSurfaceID?date=date&apikey=apikey', () => {
     expect(response.body).toEqual(testStories);
   });
 
-  it('should return 404 for non-existent url ', async () => {
+  it('should return 403 for non-existent url', async () => {
```
Comment (kschelonka) [inline servers/braze-content-proxy/src/routes/scheduledItems.integration.ts:68]: Shouldn't this be 404? also the statusCode in the test is 500

File: servers/braze-content-proxy/src/routes/scheduledItems.integration.ts [hunk 3]
```diff
@@ -65,15 +65,14 @@ describe('/scheduled-items/:scheduledSurfaceID?date=date&apikey=apikey', () => {
...
     const response = await request(app).get('/not-found');
-
-    expect(response.statusCode).toBe(404);
+    expect(response.statusCode).toBe(500);
   });
 
-  it('should return 500 if incorrect date format is provided ', async () => {
+  it('should return 400 if incorrect date format is provided ', async () => {
```
Comment (kschelonka) [inline servers/braze-content-proxy/src/routes/scheduledItems.integration.ts:73]: test statusCode is still 500

File: servers/braze-content-proxy/src/routes/scheduledItems.integration.ts [hunk 4]
```diff
@@ -83,7 +82,7 @@ describe('/scheduled-items/:scheduledSurfaceID?date=date&apikey=apikey', () => {
     );
   });
 
-  it('should return 500 if invalid api key is provided ', async () => {
+  it('should return 403 if invalid api key is provided ', async () => {
```
Comment (kschelonka) [inline servers/braze-content-proxy/src/routes/scheduledItems.integration.ts:85]: test statusCode is still 500

File: servers/braze-content-proxy/src/server.ts [hunk 5]
```diff
@@ -2,31 +2,42 @@ import express, { Application, json } from 'express';
...
   app.use((err, req, res, next) => {
     if (res.headersSent) {
       return next(err);
     }
 
+    if (err instanceof InvalidAPIKeyError || err instanceof InvalidDateError) {
+      // While I would like to return a proper error status code, Braze does not support it.
+      // So we hard code to 500
+      // https://www.braze.com/docs/user_guide/personalization_and_dynamic_content/connected_content/making_an_api_call/
+      return res.status(500).json({ error: err.message });
```
Comment (kschelonka) [inline servers/braze-content-proxy/src/server.ts:39]: probably you'll need to change this logic if you want your tests to pass as written

File: servers/braze-content-proxy/src/routes/digest.ts (hunk 1 above)
Comment (bassrock) [reply servers/braze-content-proxy/src/routes/digest.ts:46]: no its not. and its very annoying

**Review by kschelonka — APPROVED** [review]

**Pull request merged**